core/src/main/scala/com/avast/grpc/jsonbridge/GrpcJsonBridgeBase.scala

```diff
@@ -41,20 +41,32 @@ abstract class GrpcJsonBridgeBase[F[_], Stub <: io.grpc.stub.AbstractStub[Stub]]
         .to[F]
         .map(Right(_): Either[Status, A])
         .recover {
-          case e: StatusException if e.getStatus.getCode == Status.Code.UNKNOWN => Left(Status.INTERNAL)
-          case e: StatusRuntimeException if e.getStatus.getCode == Status.Code.UNKNOWN => Left(Status.INTERNAL)
-          case e: StatusException => Left(e.getStatus)
-          case e: StatusRuntimeException => Left(e.getStatus)
-          case NonFatal(e) =>
-            logger.debug("Error while executing the request", e)
-            Left(Status.INTERNAL.withCause(e))
+          case NonFatal(ex) =>
+            logger.info("Error while executing the request (recover)", ex)
+            ex match {
+              case e: StatusException if e.getStatus.getCode == Status.Code.UNKNOWN =>
+                Left(Status.INTERNAL.withCause(e.getStatus.getCause))
+              case e: StatusRuntimeException if e.getStatus.getCode == Status.Code.UNKNOWN =>
+                Left(Status.INTERNAL.withCause(e.getStatus.getCause))
+              case e: StatusException =>
+                Left(e.getStatus.withCause(e.getStatus.getCause))
```

Comment:
`e.getStatus.withCause(e.getStatus.getCause)` looks like no-op. Wouldn't following be the same? `Left(e.getStatus)`

Reply:
silly me :roll_eyes: